pkg/standalone-utils/contracts/ProtocolFeeSplitter.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/standalone-utils/IProtocolFeeSplitter.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/Authentication.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/VaultHelpers.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IProtocolFeesCollector.sol";
+
+interface Pool {
+    function getOwner() external returns (address);
+}
+
+// Packed to use 1 storage slot
+// 1e18 (100% - maximum fee value) can fit in uint96
+struct RevenueShareSettings {
+    uint96 revenueSharePercentageOverride;
+    address beneficiary;
+}
+
+/**
+ * @dev This contract is responsible for splitting the BPT profits collected
+ * by ProtocolFeeCollector between the pool's owner and Balancers DAO treasury
+ * Nothing happens for non-BPT tokens (WETH, WBTC, etc...)
+ */
+contract ProtocolFeeSplitter is IProtocolFeeSplitter, Authentication {
+    using Math for uint256;
+
+    // All fee percentages are 18-decimal fixed point numbers.
+    // Absolute maximum fee percentages (1e18 = 100%, 1e16 = 1%).
+    uint256 private constant _MIN_REVENUE_SHARING_FEE_PERCENTAGE = 1e16; // 1%
+    uint256 private constant _MAX_REVENUE_SHARING_FEE_PERCENTAGE = 50e16; // 50%
+    address private constant _DELEGATE_OWNER = 0xBA1BA1ba1BA1bA1bA1Ba1BA1ba1BA1bA1ba1ba1B;
+
+    IProtocolFeesCollector public immutable protocolFeesCollector;
+
+    // Balancer DAO Multisig
+    address public immutable treasury;
```

Comment:
Having this be `immutable` seems to be make this contract a bit too strict, surely we can make this mutable and have some future proofing?

Reply:
But if treasury changes I suppose a lot of things will change with it?
This immutability on the other hand enforces security, that DAO definitely won't be able to give an ovnership over this contract to some single address, even in the case of some intelligent attack